pulp_puppet_plugins/pulp_puppet/forge/releases.py

```diff
@@ -50,24 +83,66 @@ def view(consumer_id, repo_id, module_name, version=None):
         repo_ids = get_bound_repos(consumer_id)
     else:
         repo_ids = [repo_id]
-    if version:
-        unit = find_version(repo_ids, module_name, version)
-    else:
-        unit = find_newest(repo_ids, module_name)
-    if not unit:
-        raise web.NotFound()
+
+    dbs = None
+    return_data = None
     try:
-        data = unit.build_dep_metadata()
+        # Get the list of database files to query
+        dbs = get_repo_data(repo_ids)
+
+        # Build list of units to return
+        ret = []
+        # If a version was specified filter by that specific version of the module
+        if version:
+            for unit in unit_generator(dbs, module_name):
+                if unit.version == version:
+                    ret.append(unit)
+                    break
+        else:
+            units = [individual_unit for individual_unit in unit_generator(dbs, module_name)]
+            # if view_all_matching then return all modules matching the query, otherwise
+            # only return the first matching module (for forge v1 & v2 api compliance)
+            if view_all_matching:
+                ret = units
+            else:
+                if units:
+                    ret.append(max(units))
+
+        # calculate dependencies for the units being returned & build the return structure
+        return_data = {}
+        for unit in ret:
+            populated_unit = unit.build_dep_metadata(recurse_deps)
+            for unit_name, unit_details in populated_unit.iteritems():
+                if unit_name in return_data:
```

Comment:
FWIW, these 4 lines could be condensed to ``return_data.setdefault(unit_name, {}).extend(unit_details)``

Admittedly, I don't think the ``setdefault`` call is one of the more intuitive or elegant parts of python, but it is effective at least. It's up to you if you want to leave this code as-is.